ast2vec/tests/test_cloning.py

```diff
@@ -0,0 +1,128 @@
+            args.output = tmpdir
+            args.linguist = "xxx"
+            with self.assertRaises(FileNotFoundError):
+                clone_repositories(args)
+
+    def test_full_clone(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = tmpdir
+            args.linguist = None
+            self.assertIsNone(clone_repositories(args))
+            self._validate_clone(tmpdir, args)
+
+    def test_classified(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = tmpdir
+            args.linguist = tests.ENRY
+            self.assertIsNone(clone_repositories(args))
+            self._validate_clone(tmpdir, args)
+
+    def test_languages(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = tmpdir
+            args.linguist = tests.ENRY
+            args.languages = ["Python"]
+            self.assertIsNone(clone_repositories(args))
+            self._validate_clone(tmpdir, args)
+
+    def test_multiple_repositories(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = tmpdir
+            args.linguist = tests.ENRY
+            args.languages = ["Python"]
+            args.input.append(os.path.join(os.path.dirname(__file__), "test_repos_list.txt"))
+            self.assertIsNone(clone_repositories(args))
+            self._validate_clone(tmpdir, args)
+
+    def test_repo_creation(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = str.encode(tmpdir)
+            args.linguist = None
+            with self.assertRaises(TypeError):
+                clone_repositories(args)
+
+    def test_prepare_repo_dir(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            for repo_url in self.repo_urls:
+                repo_url = RepoCloner._prepare_repo_url(repo_url)
+                self.assertEqual(
+                    os.path.join(tmpdir, "cool@you"),
+                    RepoCloner._prepare_repo_dir(repo_url, tmpdir, True))
+
+    def test_prepare_repo_url(self):
+        results = [
+            "https://whatever.cite.com/cool/you.git",
+            "https://whatever.cite.com/cool/you.git",
+            "https://whatever.cite.com/cool/you",
+            "https://whatever.cite.com/cool/you",
+            "https://whatever.cite.com/cool/you",
+            "https://whatever.cite.com/cool/you.git",
+        ]
+        for correct_url, repo_url in zip(results, self.repo_urls):
+            self.assertEqual(correct_url, RepoCloner._prepare_repo_url(repo_url))
+
+    def _collect_repo_files(self, repo_dir):
+        files = list()
+        for dirname, dirnames, filenames in os.walk(str.encode(repo_dir)):
```

Comment:
rename dirnames to _ if you don't need this variable. [Proof](https://stackoverflow.com/questions/431866/ignore-python-multiple-return-value).
🤓